packages/ListBox/src/effects/effects.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import useListBox from "../useListBox";
+import { getDataOptions } from "../components/Option/helpers/optionState";
+
+// we should export a hook from the popover so we get notify when the popover state for isOpen change meanwhile:
+function waitForPopoverAnimation(func) {
+  setTimeout(func, 300);
+}
+
+export const handleEffectChildren = (props, state, dispatch) => () => {
+  if (Object.keys(state.options).length + state.hasFooter !== React.Children.count(props.children)) {
+    const options = getDataOptions(props.children);
+
+    dispatch({
+      type: useListBox.types.updateOptions,
+      payload: options,
+    });
+  }
+};
+
+export const handleEffectIsPopOverOpen = (state, dispatch, shouldKeepTriggerFocus) => () => {
+  if (state.isInline) return;
+  if (!state.refListBoxContainer.current) return;
+
+  const filterInput = state.refFilterInput.current;
+  const listBoxContainer = state.refListBoxContainer.current;
+  const trigger = state.refTrigger.current;
+  if (state.isOpen) {
+    if (state.hasFilter && !shouldKeepTriggerFocus) {
+      waitForPopoverAnimation(() => filterInput.focus());
+      return;
+    }
+
+    dispatch({ type: useListBox.types.setTriggerWidth, payload: state.refTriggerContainer.current.offsetWidth });
+    dispatch({ type: useListBox.types.setHasPopupOpened, payload: true });
+
+    if (!shouldKeepTriggerFocus) {
+      waitForPopoverAnimation(() => listBoxContainer.focus());
+    }
+    return;
+  }
+
+  if (state.hasPopupOpened) {
+    listBoxContainer.focus();
+    if (!state.isInline && !state.isOpen) {
+      trigger.focus();
+    }
+  }
+};
+
+export const handleEffectListBoxWidth = (state, dispatch) => () => {
+  const $triggerContainer = state.refTriggerContainer.current;
```

Comment:
I personally like the `$` prefix for DOM elements... but there are many more opportunities to use this convention consistently.  Lines 25-27 in this very file, for example.